app/main.py

```diff
@@ -2,0 +1,35 @@
+class Animal:
+    alive = []
+
+    def __init__(
+        self, name: str,
+        health: int = 100,
+        hidden: bool = False
+    ) -> None:
+        self.health = health
+        self.name = name
+        self.hidden = hidden
+        Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return (
+            f"{{Name: {self.name}, "
+            f"Health: {self.health}, "
+            f"Hidden: {self.hidden}}}"
+        )
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        if not self.hidden:
+            self.hidden = True
+        else:
+            self.hidden = False
+
+
+class Carnivore(Animal):
+    def bite(self, herbivor: Herbivore) -> None:
```

Comment:
There is a typo in the parameter name `herbivor`. It should be `herbivore` to match the class name and for consistency.